File: social-media-app/src/main/java/com/tiago_lima/social_media_app/domain/entities/Message.java
package com.tiago_lima.social_media_app.domain.entities;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "messages")
public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne
    @JoinColumn(name = "conversation_id")
    @JsonBackReference
    private Conversation conversation;

    @ManyToOne
    @JoinColumn(name = "sender_id")
    private User sender;

    private String content;
    private LocalDateTime createdAt;
    private boolean readFlag;

    @PrePersist public void onCreate() { createdAt = LocalDateTime.now(); }
}

File: social-media-app-frontend/src/models/Message.ts
import type {UserDto} from "./UserDto.ts";

export interface Message {
    id: string;
    senderId?: string;
    senderUsername?: string;
    sender?: UserDto; // prefer this when backend sends full sender
    content: string;
    createdAt: string; // ISO string
    timestamp?: string;
    readFlag: boolean;
    conversationId?: string;
}